### ISSUE TITLE
nil pointer dereference on tx sign

### ISSUE THREAD

**Issue by splix** [ethereumproject/go-ethereum#178]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5a34a6]

goroutine 90 [running]:
panic(0xbc3180, 0xc42000c0d0)
    runtime/panic.go:500 +0x1a1
math/big.(*Int).Cmp(0x0, 0xc420203580, 0x14acb20)
    math/big/int.go:314 +0x26
go-ethereum/core/types.ChainIdSigner.Equal(0xc420203580, 0xc4202b4860, 0x13ec420, 0x14acb20, 0xbf3220)
    go-ethereum/core/types/transaction_signing.go:137 +0x92
go-ethereum/core/types.(*ChainIdSigner).Equal(0xc42000df80, 0x13ec420, 0x14acb20, 0xc421ecfdc0)
    <autogenerated>:36 +0x6f
go-ethereum/core/types.Sender(0x13ec420, 0x14acb20, 0xc42cda57c0, 0x0, 0x0, 0x0, 0xb77c80, 0xc420338c00)
    go-ethereum/core/types/transaction_signing.go:87 +0x3d6
go-ethereum/core/types.(*Transaction).From(0xc42cda57c0, 0x0, 0x0, 0xc400000000, 0x13df1c0, 0xc420204a20)
    go-ethereum/core/types/transaction.go:172 +0x5b
go-ethereum/eth.(*PublicTransactionPoolAPI).subscriptionLoop(0xc420c4b9e0)
    go-ethereum/eth/api.go:962 +0x159
created by go-ethereum/eth.NewPublicTransactionPoolAPI
    go-ethereum/eth/api.go:952 +0x19f
```

on Ubuntu, started from 3.2.3

**Comment by whilei** [comment]
Resolved with #218

**Comment by splix** [comment]
Geth Classic v3.5.0

Now in different place:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5150de]

goroutine 118 [running]:
math/big.(*Int).Cmp(0xc430140000, 0x0, 0xc4277ddce0)
        /home/travis/.gimme/versions/go1.8.linux.amd64/src/math/big/int.go:314 +0x2e
github.com/ethereumproject/go-ethereum/core/types.ChainIdSigner.PublicKey(0x0, 0x3f, 0xc42a23b4a0, 0xc4225180a0, 0x16dda80, 0x17adb60, 0x16d0100, 0xc4202028c0)
        go-ethereum/core/types/transaction_signing.go:161 +0x79
github.com/ethereumproject/go-ethereum/core/types.(*ChainIdSigner).PublicKey(0x17adb60, 0xc42a23b4a0, 0x17adb60, 0xc429ec3800, 0x454d30, 0xc429ec3800, 0x0)
        <autogenerated>:34 +0x60
github.com/ethereumproject/go-ethereum/core/types.Sender(0x16dda80, 0x17adb60, 0xc42a23b4a0, 0x0, 0x0, 0x0, 0xdb38c0, 0xc420395560)
        go-ethereum/core/types/transaction_signing.go:92 +0x197
github.com/ethereumproject/go-ethereum/core/types.(*Transaction).From(0xc42a23b4a0, 0x0, 0x0, 0x0, 0x16d01c0, 0xc4202028c0)
        go-ethereum/core/types/transaction.go:172 +0x5b
github.com/ethereumproject/go-ethereum/eth.(*PublicTransactionPoolAPI).subscriptionLoop(0xc42338f080)
        go-ethereum/eth/api.go:962 +0x131
created by github.com/ethereumproject/go-ethereum/eth.NewPublicTransactionPoolAPI
        go-ethereum/eth/api.go:952 +0x180
```
